### PR TITLE
feat/header: add header, add page titles

### PR DESCRIPTION
**Pull request by Ponomaleks**

### CODE REVIEW

File: src/app/main/components/chat/chat.component.ts [hunk 1]
```diff
@@ -8,4 +12,21 @@ import { CommonModule } from '@angular/common';
...
+  ) {
+    this.chat = this.router.getCurrentNavigation()?.extras.state as ChatLastMessage;
+  }
+
+  ngOnInit() {
+    if (this.chat) {
+      this.title.setTitle(`U-PAMERS | ${this.chat.name} ${this.chat.surname} chat`);
+      this.headerService.addTitle(`${this.chat.name} ${this.chat.surname}`);
+    }
+  }
```
Comment (GennadiyShpak) [inline src/app/main/components/chat/chat.component.ts:31]: incapsulate it to independent function

File: src/app/main/components/contact/contact.component.ts [hunk 2]
```diff
@@ -8,4 +12,21 @@ import { CommonModule } from '@angular/common';
...
+    private headerService: HeaderService
+  ) {
+    this.user = this.router.getCurrentNavigation()?.extras.state as UserDetailed;
+  }
+
+  ngOnInit() {
+    if (this.user) {
+      this.title.setTitle(`U-PAMERS | ${this.user.name} ${this.user.surname} profile`);
+      this.headerService.addTitle(`${this.user.name} ${this.user.surname}`);
+    }
```
Comment (GennadiyShpak) [inline src/app/main/components/contact/contact.component.ts:30]: this too

File: src/app/shared/components/epm-header/epm-header.component.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    <img
+      src="{{ header_icons[headerSettings().leftButton].src }}"
+      height="{{ header_icons[headerSettings().leftButton].size }}"
+      width="{{ header_icons[headerSettings().leftButton].size }}"
+      [alt]="header_icons[headerSettings().leftButton].alt"
+    />
+  </a>
+  <h1 class="title">{{ headerSettings().title }}</h1>
+
+  <ng-container *ngIf="loggedId; then authorized; else unauthorized"></ng-container>
```
Comment (GennadiyShpak) [inline src/app/shared/components/epm-header/epm-header.component.html:16]: what is loggedId?

File: src/app/shared/components/epm-header/epm-header.component.scss [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    align-items: center;
+
+    .left-icon {
+    }
+
+    .title {
+      position: absolute;
+      left: 50%;
+      transform: translateX(-50%);
+      font-size: 16px;
```
Comment (GennadiyShpak) [inline src/app/shared/components/epm-header/epm-header.component.scss:26]: why do you nedd style nesting in host?

File: tsconfig.json [hunk 5]
```diff
@@ -7,7 +7,7 @@
     "forceConsistentCasingInFileNames": true,
     "strict": true,
     "noImplicitOverride": true,
-    "noPropertyAccessFromIndexSignature": true,
+    "noPropertyAccessFromIndexSignature": false,
```
Comment (GennadiyShpak) [inline tsconfig.json:10]: why?

File: src/app/shared/components/epm-header/epm-header.component.html (hunk 3 above)
Comment (Ponomaleks) [reply src/app/shared/components/epm-header/epm-header.component.html:16]: Than should be loggedI**n**. But I will rename it to isAuthorized

File: tsconfig.json (hunk 5 above)
Comment (Ponomaleks) [reply tsconfig.json]: changed back

File: src/app/shared/components/epm-header/epm-header.component.scss (hunk 4 above)
Comment (Ponomaleks) [reply src/app/shared/components/epm-header/epm-header.component.scss]: changed

File: src/app/main/components/chat/chat.component.ts (hunk 1 above)
Comment (Ponomaleks) [reply src/app/main/components/chat/chat.component.ts:35]: done

File: src/app/main/components/contact/contact.component.ts (hunk 2 above)
Comment (Ponomaleks) [reply src/app/main/components/contact/contact.component.ts:34]: done

File: src/app/app.model.ts [hunk 6]
```diff
@@ -1,3 +1,25 @@
+import { HEADER_ICON_TYPE, HEADER_RIGHT_BLOCK } from './app.config';
+
+export interface HeaderSettings {
+  leftButton: HEADER_ICON_TYPE;
+  title?: string;
+  right?: HEADER_RIGHT_BLOCK;
+}
+
+export type headerSettingsType =
```
Comment (GennadiyShpak) [inline src/app/app.model.ts:9]:
```suggestion
export type HeaderSettingsType =
```

File: src/app/auth/container/auth.routes.ts [hunk 7]
```diff
@@ -16,19 +16,23 @@ export default [
       },
       {
         path: APP_ROUTER_NAME.LogIn,
-        component: LogInComponent
+        component: LogInComponent,
+        data: headerSettingsSet['logIn']
```
Comment (GennadiyShpak) [inline src/app/auth/container/auth.routes.ts:20]: Why?

File: src/app/main/components/contact/contact.component.ts [hunk 8]
```diff
@@ -8,4 +12,25 @@ import { CommonModule } from '@angular/common';
   templateUrl: './contact.component.html',
   styleUrls: ['./contact.component.scss']
 })
-export class ContactComponent {}
+export class ContactComponent implements OnInit {
+  private readonly user!: UserDetailed;
```
Comment (GennadiyShpak) [inline src/app/main/components/contact/contact.component.ts:16]: wtf?

File: src/app/services/header.service.ts [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+      )
+      .subscribe();
+  }
+
+  addTitle(title: string): void {
+    this.headerSettings$.mutate(settings => (settings.title = title));
+  }
+
+  back(): void {
+    const previousRoutePage: string[] = this.history[this.history.length - 2]?.split('/');
```
Comment (GennadiyShpak) [inline src/app/services/header.service.ts:54]: .at(-2)

File: src/app/shared/components/epm-header/epm-header.component.html [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+          />
+        </div>
+        <a class="avatar-wrapper" [routerLink]="myProfileLink">
+          <img width="24" height="24" [ngSrc]="avatar.src" alt="user avatar" />
+        </a>
+      </div>
+    </ng-container>
+
+    <ng-container *ngSwitchCase="header_right_block.Skip">
+      <span class="skip">Skip</span>
```
Comment (GennadiyShpak) [inline src/app/shared/components/epm-header/epm-header.component.html:25]: Why span?

File: src/app/shared/components/epm-header/epm-header.component.ts [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+  isAuthorized = true;
+  avatar = { src: '/mocks/img/avatar.png' };
+
+  newMessage: Signal<boolean> = signal(true);
+  headerSettings: Signal<HeaderSettings> = this.headerService.headerSettings;
+
+  readonly myProfileLink: string = `${APP_ROUTER_NAME.Main}/${APP_ROUTER_NAME.MyProfile}`;
+  readonly chatLink: string = `${APP_ROUTER_NAME.Main}/${APP_ROUTER_NAME.Chat}`;
+  readonly button_themes = BUTTON_THEMES;
+  readonly header_right_block = HEADER_RIGHT_BLOCK;
```
Comment (GennadiyShpak) [inline src/app/shared/components/epm-header/epm-header.component.ts:29]: type?

File: src/app/shared/components/epm-header/epm-header.component.ts [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+
+  handleLogIn(): void {
+    // do nothing
+  }
+
+  back(): void {
+    this.headerService.back();
+  }
+
+  protected readonly HEADER_ICON_TYPE = HEADER_ICON_TYPE;
```
Comment (GennadiyShpak) [inline src/app/shared/components/epm-header/epm-header.component.ts:45]: why here?

File: src/app/auth/container/auth.routes.ts (hunk 7 above)
Comment (Ponomaleks) [reply src/app/auth/container/auth.routes.ts:20]: I don't see the problem

File: src/app/app.model.ts (hunk 6 above)
Comment (Ponomaleks) [reply src/app/app.model.ts:9]: changed to enum HEADER_CONFIG_NAME

File: src/app/auth/container/auth.routes.ts (hunk 7 above)
Comment (Ponomaleks) [reply src/app/auth/container/auth.routes.ts:20]: refactored to HEADER_CONFIG_LIST with keys from enum

File: src/app/main/components/contact/contact.component.ts (hunk 8 above)
Comment (Ponomaleks) [reply src/app/main/components/contact/contact.component.ts:16]: changed

File: src/app/services/header.service.ts (hunk 9 above)
Comment (Ponomaleks) [reply src/app/services/header.service.ts:54]: thanks. Used .at(-2)

File: src/app/shared/components/epm-header/epm-header.component.ts (hunk 11 above)
Comment (Ponomaleks) [reply src/app/shared/components/epm-header/epm-header.component.ts:29]: added

File: src/app/shared/components/epm-header/epm-header.component.ts (hunk 12 above)
Comment (Ponomaleks) [reply src/app/shared/components/epm-header/epm-header.component.ts:45]: remuved redundant variable

File: src/app/shared/components/epm-header/epm-header.component.html (hunk 10 above)
Comment (Ponomaleks) [reply src/app/shared/components/epm-header/epm-header.component.html]: Changed to epm-button

**Review by GennadiyShpak — APPROVED** [review]

**Pull request merged**